### DISCUSSION TITLE
What is the purpose of the name on the front of `states`

### DISCUSSION THREAD

**Discussion by shanliu** [slint-ui/slint#4636] [Q&A] [answered]

**Comment by ogoffart** [comment, marked as answer]
Thanks for your question.
(I moved it from issues to discussions because issues are for bug reports)

Currently there is no purpose other than giving it a name for the developer to know what's the meening of the state.

In the previous syntax, it was used to define transitions, but now the syntax don't need to repeat the state name.

In the future, we could have state inheritance (https://github.com/slint-ui/slint/issues/2089) or substates or sequence of states (https://github.com/slint-ui/slint/issues/2255) in which case it will be usefull to be able to specify the state by name.

**Reply by kanru** [reply]
It would be useful if elements can reference the states, so no need to use another state property.

For example

```
if states.show-table: VerticalBox {
 // ...
}
```

instead of

```
if state == TableState.show-table: VerticalBox {
}
```
